dhlparcel/keycloak-metrics-spi
# Keycloak 19 fix realm name

## Motivation
In Keycloak 19 the 'realm' in the metrics contain a UUID instead of the Realm name.
Apparently Keycloak changed the value of `event.getRealmId()` from a Realm name to a real Realm id. 

## What
This change will fix this by looking up the Realm by Id via a RealmProvider so the metrics will still contain a Realm name.

Upgraded Keycloak dependency to 19.0.1 to also verify this library is not using any deprecated sources.
Removed the `MetricsEventListenerSpi` since this seems to be an old way of adding a listener and is not needed anymore for v19 Quarkus Keycloak.

## Why
To have readable realm names in the metrics and on the Keycloak dashboard.

## How
Realm name is looked up by using a RealmProvider which is accessible from the KeycloakSession when the MetricsEventListenerFactory is created.

## Verification Steps
1. Build the package, include it in a custom Keycloak 19+ image and start the container.
2. Login into the admin console
3. Enable metrics-listener for events in the Realm settings.
4. Logout
5. Log back into admin console
6. Run `curl -vk https://<keycloak-url>:<port>/auth/realms/master/metrics | grep -v '^#' | grep login` and check metrics output.
The metrics should still contain the realm name like
```
keycloak_logins{realm="admin",provider="keycloak",client_id="account-console",} 1.0
```

## Checklist:


## Progress
